src/app/personalised-homepage/page.js

```diff
@@ -0,0 +1,72 @@
+'use client'
+import Aside from "../components/personalised-homepage-components/Aside";
+import ItemCard from "../components/personalised-homepage-components/ItemCard";
+import ItemContainer from "../components/personalised-homepage-components/ItemContainer";
+import ItemRow from "../components/personalised-homepage-components/ItemRow";
+import ItemRowContainer from "../components/personalised-homepage-components/ItemRowContainer";
+import Bg from "../components/personalised-homepage-components/Bg";
+import { useEffect, useState } from "react";
+import { fetchArtistAlbum, fetchProfile, fetchUserPlaylists, fetchUserTracks } from "../hooks/spotify/spotify_hooks";
+
+function PersonalisedHomepage() {
+    let accessToken = "";
+
+    if (typeof window !== "undefined") {
+        accessToken = window.location.hash.split("&")[0].split("=")[1];
+    }
+
+    const [profileName, setProfileName] = useState("");
+    const [userImage, setUserImage] = useState("");
+    const [albums, setAlbums] = useState([]);
+    const [tracks, setTracks] = useState([]);
+    const [playlists, setPlaylists] = useState([]);
+
+    useEffect(() => {
+        fetchProfile(accessToken).then((res) => setUserEmail(res.email));
+        fetchProfile(accessToken).then((res) => setProfileName(res.display_name));
+        fetchProfile(accessToken).then((res) => setUserImage(res.images?.[1]?.url));
+        fetchArtistAlbum(accessToken).then((res) => setAlbums(res.items));
+        fetchUserTracks(accessToken).then((res) => setTracks(res.items));
+        fetchUserPlaylists(accessToken).then((res) => setPlaylists(res.playlists?.items));
+    }, []);
+
+    // Useful data for ML / backend //
+
+    // User top tracks is an array of users top played track IDs
+    const userTopTracks = tracks?.map((track) => {
+        return track.id
+    })
+
+    // User Email - could be used as a unique identifier to store in DB and match to personality type
+    const [userEmail, setUserEmail] = useState("");
+
+    console.log(userEmail)
+
+    return (
+        <Bg>
+            <Aside username={profileName} personalityType="Architect" profileImage={userImage}>
+            </Aside>
+            <ItemContainer>
+                <ItemRowContainer>
+                    <ItemRow title="Recommended Songs">
+                        {tracks?.map((track, index) => {
+                            return <ItemCard key={`song${index}`} title={track.name} img={track.album.images[1].url}>{track.artists[0].name}</ItemCard>
```

Comment:
**Opinionated view**:`images[1]`, `artists[0]`. 

As a code reviewer, and possibly your future self, we have no clue why these indexes where chosen.

To possibly improve readability and maintainability [**not necessary**], we could store these in a meaningful variable, allowing our future selves to come back to this snippet and quickly understand what indexes 0 and 1 mean in these arrays.

Whenever I see a number, I am always tempted to store its value in a variable that gives it meaning.

Not directly relevant, just an opportunity to share a resource 🤣 : https://github.com/ryanmcdermott/clean-code-javascript#use-searchable-names

Reply:
Gotchya! The images numbers relate to size, so I will store these numbers in a variable outlining the size of image used. Thanks for this.